app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.java

```diff
@@ -121,73 +121,81 @@ public ViewModel(final @NonNull Environment environment) {
-      final Observable<RefTag> cookieRefTag = this.project
-        .take(1)
-        .map(p -> RefTagUtils.storedCookieRefTagForProject(p, this.cookieManager, this.sharedPreferences));
-
       final Observable<Project> initialProject = intent()
         .flatMap(i -> ProjectIntentMapper.project(i, this.client))
         .share();
 
+      // An observable of the ref tag stored in the cookie for the project. Can emit `null`.
+      final Observable<RefTag> cookieRefTag = initialProject
+        .take(1)
+        .map(p -> RefTagUtils.storedCookieRefTagForProject(p, this.cookieManager, this.sharedPreferences));
+
       final Observable<RefTag> refTag = intent()
         .flatMap(ProjectIntentMapper::refTag);
 
       final Observable<PushNotificationEnvelope> pushNotificationEnvelope = intent()
         .flatMap(ProjectIntentMapper::pushNotificationEnvelope);
 
       final Observable<User> loggedInUserOnStarClick = this.currentUser.observable()
-        .compose(takeWhen(this.starButtonClickedSubject))
+        .compose(takeWhen(this.starButtonClicked))
         .filter(u -> u != null);
 
       final Observable<User> loggedOutUserOnStarClick = this.currentUser.observable()
-        .compose(takeWhen(this.starButtonClickedSubject))
+        .compose(takeWhen(this.starButtonClicked))
         .filter(u -> u == null);
 
       final Observable<Project> projectOnUserChangeStar = initialProject
         .compose(takeWhen(loggedInUserOnStarClick))
         .switchMap(this::toggleProjectStar)
         .share();
 
-      final Observable<Project> starredProjectOnLoginSuccess = this.showLoginTout
+      this.startLoginToutActivity = loggedOutUserOnStarClick.compose(ignoreValues());
+
+      final Observable<Project> starredProjectOnLoginSuccess = this.startLoginToutActivity
         .compose(combineLatestPair(this.currentUser.observable()))
         .filter(su -> su.second != null)
         .withLatestFrom(initialProject, (__, p) -> p)
         .take(1)
         .switchMap(this::starProject)
         .share();
 
-      initialProject
-        .mergeWith(projectOnUserChangeStar)
-        .mergeWith(starredProjectOnLoginSuccess)
-        .compose(bindToLifecycle())
-        .subscribe(this.project::onNext);
-
-      projectOnUserChangeStar.mergeWith(starredProjectOnLoginSuccess)
-        .filter(Project::isStarred)
-        .filter(Project::isLive)
-        .filter(p -> !p.isApproachingDeadline())
-        .compose(bindToLifecycle())
-        .subscribe(__ -> this.showStarredPrompt.onNext(null));
-
-      loggedOutUserOnStarClick
-        .compose(bindToLifecycle())
-        .subscribe(__ -> this.showLoginTout.onNext(null));
-
-      this.shareButtonClickedSubject
+      final Observable<Project> currentProject = Observable.merge(
+        initialProject,
+        projectOnUserChangeStar,
+        starredProjectOnLoginSuccess
+      );
+
+      this.showStarredPrompt = projectOnUserChangeStar.mergeWith(starredProjectOnLoginSuccess)
```

Comment:
it should still be showing, maybe it was a project ending within 48 hours?

we don't show the toast for that since it says `"We'll remind you 48 hours before this project ends!"` which also makes me realize that we should add logic in the future to show a different toast for that case